src/manifest.ts

```diff
@@ -0,0 +1,176 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {GitHub, GitHubFileContents} from './github';
+import {ReleaseType} from './releasers';
+import {RELEASE_PLEASE_CONFIG, RELEASE_PLEASE_MANIFEST} from './constants';
+import {ManifestConstructorOptions} from '.';
+import {ChangelogSection} from './conventional-commits';
+import {CheckpointType, checkpoint, Checkpoint} from './util/checkpoint';
+import {GitHubReleaseResponse} from './github-release';
+
+interface ReleaserConfig {
+  'release-type'?: ReleaseType;
+  'bump-minor-pre-major'?: boolean;
+  'changelog-sections'?: ChangelogSection[];
+}
+
+interface ReleaserPackageConfig extends ReleaserConfig {
+  'package-name'?: string;
+}
+
+export interface ManifestConfig extends ReleaserConfig {
+  packages: Record<string, ReleaserPackageConfig>;
+}
+
+export class Manifest {
+  gh: GitHub;
+  configFileName: string;
+  manifestFileName: string;
+  checkpoint: Checkpoint;
+  configFile?: GitHubFileContents;
+  headManifest?: GitHubFileContents;
+
+  constructor(options: ManifestConstructorOptions) {
+    this.gh = options.github;
+    this.configFileName = options.configFile || RELEASE_PLEASE_CONFIG;
+    this.manifestFileName = options.manifestFile || RELEASE_PLEASE_MANIFEST;
+    this.checkpoint = options.checkpoint || checkpoint;
+  }
+
+  protected async getFile(
+    fileName: string,
+    sha?: string
```

Comment:
This `sha?` parameter and treatment really belong to the next PR but I didn't want to do too much git surgery so I left it in here.